File: 1752. Check if Array Is Sorted and Rotated/Solution.py
class Solution(object):
    def check(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        count = 0
        n = len(nums)

        for i in range(n):
            if nums[i] > nums[(i + 1) % n]:
                count += 1
                if count > 1:
                    return False

        return True
    

#another solution     
# class Solution(object):
#     def check(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: bool
#         """
#         n = len(nums)

#         # find index of minimum element
#         min_index = 0
#         for i in range(1, n):
#             if nums[i] < nums[min_index]:
#                 min_index = i

#         # check sorted order in circular manner
#         for i in range(n - 1):
#             curr = nums[(min_index + i) % n]
#             nxt = nums[(min_index + i + 1) % n]
#             if curr > nxt:
#                 return False

#         return True

File: 1752. Check if Array Is Sorted and Rotated/Solution.java
class Solution {
    public boolean check(int[] nums) {
        int count = 0;
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            if (nums[i] > nums[(i + 1) % n]) {
                count++;
                if (count > 1) {
                    return false;
                }
            }
        }
        return true;
    }
}


//another solution 
// class Solution {
//     public boolean check(int[] nums) {
//         int n = nums.length;

//         int minIndex = 0;
//         for (int i = 1; i < n; i++) {
//             if (nums[i] < nums[minIndex]) {
//                 minIndex = i;
//             }
//         }

//         // check if array is sorted from minIndex in circular way
//         for (int i = 0; i < n - 1; i++) {
//             int curr = nums[(minIndex + i) % n];
//             int next = nums[(minIndex + i + 1) % n];
//             if (curr > next) {
//                 return false;
//             }
//         }

//         return true;
//     }
// }
